MAINT: ndimage test failures with CuPy

These look pretty new, I see 240 failures locally on x86_64 Linux via `python dev.py test -j 32 -b all -t scipy/ndimage/tests`. I'm not sure this is surprising based on the discussion in gh-20809, but I'm using CuPy `cupy-cuda12x` `13.2.0`.  Bumping up to `13.3.0` (latest stable) has no effect either.

This causes the problem to go away, presumably by skipping CuPy, though perhaps too aggressively:

```diff
--- a/scipy/ndimage/tests/test_filters.py
+++ b/scipy/ndimage/tests/test_filters.py
@@ -25,7 +25,7 @@ skip_xp_backends = pytest.mark.skip_xp_backends
 xfail_xp_backends = pytest.mark.xfail_xp_backends
 pytestmark = [array_api_compatible, pytest.mark.usefixtures("skip_xp_backends"),
               pytest.mark.usefixtures("xfail_xp_backends"),
-              skip_xp_backends(cpu_only=True, exceptions=['cupy', 'jax.numpy']),]
+              skip_xp_backends(cpu_only=True, exceptions=['jax.numpy']),]
```

To state the obvious, that block is pretty hard to read. Anyway, could we skip these tests or something for now, that's a lot of noise in the array API suite.

I'll paste a representative failure below, and proposed that we skip for now in the cross-linked PR.

<details>

```
_____________________________________________________ TestNdimageFilters.test_correlate_convolve_axes[origin1-axes14-float64-correlate-cupy] ______________________________________________________
[gw23] linux -- Python 3.11.2 /home/treddy/python_venvs/py_311_scipy_dev/bin/python
scipy/ndimage/tests/test_filters.py:943: in test_correlate_convolve_axes
    output = func(array, weights=weights, axes=axes, origin=origin)
        array      = array([[[  0.,   1.,   2.,   3.,   4.,   5.,   6.,   7.,   8.,   9.,
          10.,  11.],
        [ 12.,  13.,  14., ...61.,
         562., 563.],
        [564., 565., 566., 567., 568., 569., 570., 571., 572., 573.,
         574., 575.]]])
        axes       = (1, 2)
        dtype      = <class 'numpy.float64'>
        func       = <function correlate at 0x7fc81dfd0cc0>
        origin     = (-1, 1)
        self       = <scipy.ndimage.tests.test_filters.TestNdimageFilters object at 0x7fc8143496d0>
        weights    = array([[ 0,  1,  2,  3,  4],
       [ 5,  6,  7,  8,  9],
       [10, 11, 12, 13, 14]])
        xp         = <module 'cupy' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupy/__init__.py'>
scipy/ndimage/_support_alternative_backends.py:28: in wrapper
    return cupyx_func(*args, **kwds)
E   TypeError: correlate() got an unexpected keyword argument 'axes'
        args       = (array([[[  0.,   1.,   2.,   3.,   4.,   5.,   6.,   7.,   8.,   9.,
          10.,  11.],
        [ 12.,  13.,  14.,....,
         562., 563.],
        [564., 565., 566., 567., 568., 569., 570., 571., 572., 573.,
         574., 575.]]]),)
        cupyx_func = <function correlate at 0x7fc8127d5940>
        cupyx_module = <module 'cupyx.scipy.ndimage' from '/home/treddy/python_venvs/py_311_scipy_dev/lib/python3.11/site-packages/cupyx/scipy/ndimage/__init__.py'>
        dispatcher = <function convolve_dispatcher at 0x7fc81dfaed40>
        func       = <function correlate at 0x7fc81df7e840>
        importlib  = <module 'importlib' from '/home/treddy/github_projects/spack/opt/spack/linux-ubuntu22.04-skylake/gcc-11.3.0/python-3.11.2-4qncg2nqev5evxfmamde3e6rnb34b4ls/lib/python3.11/importlib/__init__.py'>
        kwds       = {'axes': (1, 2), 'origin': (-1, 1), 'weights': array([[ 0,  1,  2,  3,  4],
       [ 5,  6,  7,  8,  9],
       [10, 11, 12, 13, 14]])}
        module_name = 'ndimage'
        xp         = <module 'scipy._lib.array_api_compat.cupy' from '/home/treddy/github_projects/scipy/build-install/lib/python3.11/site-packages/scipy/_lib/array_api_compat/cupy/__init__.py'>
```

</details>

--- a/scipy/ndimage/tests/test_filters.py
+++ b/scipy/ndimage/tests/test_filters.py
@@ -922,6 +922,8 @@ class TestNdimageFilters:
         expected = filter_func(array, *extra_args, all_sizes)
         xp_assert_close(output, expected)
 
+    @skip_xp_backends("cupy",
+                      reasons=["https://github.com/cupy/cupy/pull/8339"])
     @pytest.mark.parametrize('func', [ndimage.correlate, ndimage.convolve])
     @pytest.mark.parametrize(
         'dtype', [np.float32, np.float64, np.complex64, np.complex128]

